public/components/datasources/components/manage/associated_objects_tab.tsx

```diff
@@ -122,40 +136,119 @@ export const AssociatedObjectsTab: React.FC<AssociatedObjectsTabProps> = ({
+        return accelerations.length > 0
+          ? accelerations.map((acceleration, index) => (
+              <React.Fragment key={index}>
+                <EuiLink onClick={() => openFlyout(acceleration)}>{acceleration}</EuiLink>
+                {index < accelerations.length - 1 ? ', ' : ''}
+              </React.Fragment>
+            ))
+          : '-';
+      },
     },
     {
       name: 'Actions',
       actions: [
         {
-          name: 'Edit',
-          description: 'Edit this object',
+          name: 'Discover',
+          description: 'Discover this object',
           type: 'icon',
           icon: 'discoverApp',
-          onClick: (item: AssociatedObject) => console.log('Edit', item),
+          onClick: (item: AssociatedObject) => console.log('Discover', item),
         },
         {
-          name: 'Delete',
-          description: 'Delete this object',
+          name: 'Accelerate',
+          description: 'Accelerate this object',
           type: 'icon',
           icon: 'bolt',
-          onClick: (item: AssociatedObject) => console.log('Delete', item),
+          available: (item: AssociatedObject) => item.type === 'Table',
+          onClick: (item: AssociatedObject) => console.log('Accelerate', item),
         },
       ],
     },
   ];
 
+  const onSearchChange = ({ query, error }) => {
+    if (error) {
+      console.log('Search error:', error);
+      return;
+    }
+
+    const matchesClauses = (obj, clauses) => {
+      if (!clauses.length) {
+        return true;
+      }
+
+      return clauses.some((clause) => {
+        if (clause.type === 'field') {
+          if (clause.field === 'accelerations' && Array.isArray(obj[clause.field])) {
+            return obj[clause.field].includes(clause.value);
+          } else {
+            switch (clause.operator) {
+              case 'eq':
+                return obj[clause.field] === clause.value;
+              default:
+                return true;
+            }
+          }
+        }
+        return true;
+      });
+    };
+
+    const filtered = associatedObjects.filter((obj) => {
+      const clauses = query.ast._clauses;
+      return matchesClauses(obj, clauses);
+    });
+
+    setFilteredObjects(filtered);
+  };
+
+  const searchFilters = [
+    {
+      type: 'field_value_selection',
+      field: 'database',
+      name: 'Database',
+      multiSelect: true,
+      options: databaseFilterOptions,
+      cache: 60000,
+      onChange: (value) => setSelectedDatabaseFilter(value),
+    },
+    {
+      type: 'field_value_selection',
+      field: 'accelerations',
+      name: 'Accelerations',
+      multiSelect: true,
+      options: accelerationFilterOptions,
+      cache: 60000,
+      onChange: (value) => setSelectedAccelerationFilter(value),
+    },
+  ];
+
   const search = {
+    filters: searchFilters,
     box: {
       incremental: true,
+      placeholder: 'database:database_1 accelerations:skipping_index_1',
```

Comment:
Could you also move this to a constant file

Reply:
done.